### ISSUE TITLE
Simulation output for an endogenous switching DSGE model

### ISSUE THREAD

**Issue by yukimrkm** [jmaih/RISE_toolbox#182]
Hello everyone,

I'm sorry for bothering you. I have a question about the simulation output for a regime-switching DSGE model with endogenous transition probabilities.

Simulating an endogenous regime-switching DSGE model with `simulate()` returns `mcName_tp_i_j` and `mcName_i_j` where `mcName` is the name of the Markov process. 
 
I thought both should correspond to the off-diagonal elements of the simulated transition matrix at each period, but their values do not coincide.

Specifically, the simulated value of `mcName_i_j` at each period t is exactly the transition probability evaluated at x_{t-1}, where x_{t} is an endogenous variable the transition probability is a function of, whereas simulated values for `mcName_tp_i_j` are something different. 
Only the difference I have noticed so far is that `mcName_tp_i_j` is an endogenous variable that I declared, while the other is not.

Is it normal to get different values for these two? If so, what is the difference between the two?

Thank you very much in advance for your time and kindness.
Yuki

**Comment by jmaih** [comment]
Hi Yuki,1-) They should be the same in theory because they represent the same object.2-) In practice they are not because of the solution method used. 3-) To understand this, remember that the model is solved via perturbation. Hence, all the policy functions are approximated up to the order of perturbation. In particular then, this will also be the case for the endogenous probabilities.4-) To clear the confusion, the transition probabilities are used in two different ways. First, they enter the transition matrix and there, they are not approximated. They give the exact values as you found out. Second, they enter the very formulation of system of stochastic difference equations to solve, the problem for which we do not know the exact solution.5-) Back to the theory, to the extent that the perturbation solution converges to the exact solution as the order of approximation increases, the two elements should be equal.6-) now is there any use for the approximation? Yes there is! The policy functions tell us something about how the endogenous probabilities respond to state variables. An information that the exact values would not provide.Cheers,J. Sendt fra min iPhone8. nov. 2023 kl. 17:37 skrev yuki.murakami ***@***.***>:﻿
Hello everyone,
I'm sorry for bothering you. I have a question about the simulation output for a regime-switching DSGE model with endogenous transition probabilities.
Simulating an endogenous regime-switching DSGE model with simulate() returns mcName_tp_i_j and mcName_i_j where mcName is the name of the Markov process.
I thought both should correspond to the off-diagonal elements of the simulated transition matrix at each period, but their values do not coincide.
Specifically, the simulated value of mcName_i_j at each period t is exactly the transition probability evaluated at x_{t-1}, where x_{t} is an endogenous variable the transition probability is a function of, whereas simulated values for mcName_tp_i_j are something different.
Only the difference I have noticed so far is that mcName_tp_i_j is an endogenous variable that I declared, while the other is not.
Is it normal to get different values for these two? If so, what is the difference between the two?
Thank you very much in advance for your time and kindness.
Yuki
